pkg/process/checks/container_rt.go

```diff
@@ -42,8 +43,16 @@ func (r *RTContainerCheck) Init(_ *SysProbeConfig, hostInfo *HostInfo) error {
 
 // IsEnabled returns true if the check is enabled by configuration
 func (r *RTContainerCheck) IsEnabled() bool {
-	// TODO - move config check logic here
-	return true
+	// The process and container checks are mutually exclusive
+	if ddconfig.Datadog.GetBool("process_config.process_collection.enabled") || !ddconfig.IsAnyContainerFeaturePresent() {
+		return false
+	}
+
+	var (
+		containerCollectionEnabled = ddconfig.Datadog.GetBool("process_config.container_collection.enabled")
```

Comment:
The realtime container check logic here essentially duplicates the logic from the container check logic and then adds the check for `process_config.disable_realtime_checks`. Thoughts on refactoring this, or at least leaving a comment that both places need to be updated if the logic changes?

Reply:
Good question. I'm a little weary about depending on the `IsEnabled` methods from other checks. I think if we're not careful, it could quickly become difficult to reason about the logic for a couple reasons:
- Dependency loops are something that we need to be concerned about here. IMO someone who is maintaining these `IsEnabled` methods shouldn't really have to reason about a dependency graph of `IsEnabled` functions.
- To call another check's `IsEnabled` function you must either construct a new instance of the check or ask `fx` for it's existing reference of that check. Constructing a new instance of the check isn't a great solution here, because there's a possibility that the `IsEnabled` function depends on struct fields that aren't available in all checks. Additionally, asking `fx` for the existing instance of the instantiated check creates the issue of dependency loops, as two components cannot depend on each other.

My opinion is that we should just leave a comment here, explaining that they should have similar logic, and leave it up to the maintainer to update both.

Reply:
Actually, I could also just refactor this logic into a helper function 😅 I just updated it! My point above was primarily that I want to avoid something like this:
```go
func (c *RTContainerCheck) IsEnabled() {
  return containerCheck.IsEnabled() && realtimeChecksEnabled
}
```